### DISCUSSION TITLE
[IntelliSense] syntax highlighting for tailwind class names

### DISCUSSION THREAD

**Discussion by frank-swp** [tailwindlabs/tailwindcss#10093] [Ideas]
Tailwind class names looking currently like a normal string in VS Code.

* this makes it difficult to distinguish from other class names / or normal strings in the html code
* you only know if if a tailwind class is a correct spelled if you hover it (or use a linter)

highlighting valid tailwind classname inside the a string with a special color could improve the usability.

current:
![image](https://user-images.githubusercontent.com/64835150/207873582-825299c5-9ad0-4dd7-81cc-868e90e93b64.png)

mockup:

![image](https://user-images.githubusercontent.com/64835150/207874989-319bc692-2aee-49d0-96e9-ae52f1435e57.png)

I think this would make it easier for the eye to distinguish between tailwind styles and any other string in the code.

**Comment by ProPhycient5** [comment]
Yeah @frank-swp, this is a good idea for building a new extension on `VScode` editor. Basically, you expect the color to distinguish `TailwindCSS` library classname and customized classnames that are created by the developers. If such an extension is released, then it will definitely add the next level of aesthetic value to the `JSX` code.

**Comment by ruanswanepoel** [comment]
This would be pretty helpful. 

Even being able to distinguish between tailwind classes and custom css classes would be helpful

**Reply by ProPhycient5** [reply]
Exactly @ruanswanepoel

**Comment by finnmerlett** [comment]
All the tailwind vscode extension needs to do is set textmate scopes on the identified tailwind classes, then you could set custom highlighting however you wanted

**Comment by schoero** [comment]
I created something today: [vscode-tailwindcss-syntax-highlighting](https://marketplace.visualstudio.com/items?itemName=schoero.vscode-tailwindcss-syntax-highlighting)

It's still in the early stages, so I'd really appreciate any feedback or suggestions. Please feel free to try it out and let me know what you think 🙌

**Reply by jpike88** [reply]
thanks, im using it now!

**Reply by provokcx** [reply]
@schoero I tested vscode-tailwindcss-syntax-highlighting. 
It doesn't work: 
- in class functions like cva, clix, etc. 
- in react components class attributes

**Comment by gustavopch** [comment]
Would be also nice to use different colors for different types of classes, for example:
- Classes that affect position and display are gray
- Classes that affect margin and padding are green
- Classes that affect text and font are pink
- Etc.
Ideally, the colors should not be the same as the ones used in the rest of the code, so CSS classes don't look like HTML attributes.
